src/OpenXR/Silk.NET.OpenXR/XR.cs

```diff
@@ -16,17 +17,18 @@ public partial class XR
         public Instance? CurrentInstance { get; set; }
         public static XR GetApi()
         {
-             var ret = new XR(CreateDefaultContext(new OpenXRLibraryNameContainer().GetLibraryName()));
-             return ret;
+            return new XR(CreateDefaultContext(new OpenXRLibraryNameContainer().GetLibraryName()));
         }
 
         [Obsolete("Use IsInstanceExtensionPresent instead.", true)]
         public override bool IsExtensionPresent(string extension) => IsInstanceExtensionPresent(null, extension);
-        private Dictionary<string, List<string>> _cachedInstanceExtensions = new Dictionary<string, List<string>>();
+        private readonly HashSet<string> _cachedInstanceExtensions = new HashSet<string>();
+        private readonly ReaderWriterLockSlim _cachedInstanceExtensionsLock = new ReaderWriterLockSlim();
```

Comment:
Oh by the way make sure this will work for multiple instances as my changes as part of #180 will add better support for multiple instances.

Reply:
`BreakneckLock` doesn't support read upgrading.  Further, the write is _very_ slow compared to the read, so a spinlock may not be the best idea (even a very fast one).  `ReaderWriterLockSlim` is effectively a fast spinlock on read, unless there is a pending (slow write) in which case it will yield.  It's designed for this asymmetric load.

Reply:
Fair enough. I'm still not convinced it's entirely as slim as it could be, given it's a BCL type there's got to be some safety we can rip out, but this is good for now.

Reply:
> Oh by the way make sure this will work for multiple instances as my changes as part of #180 will add better support for multiple instances.

Yes, even though it's a single hashset it's keyed as '<instance>|<extension>'